core/shared/src/main/scala/org/http4s/headers/Cookie.scala

```diff
@@ -31,12 +31,13 @@ object Cookie {
     ParseResult.fromParser(parser, "Invalid Cookie header")(s)
 
   private[http4s] val parser: Parser[Cookie] = {
-    import Parser.{char, string}
+    import Parser.char
 
     /* cookie-string = cookie-pair *( ";" SP cookie-pair ) */
-    val cookieString = (RequestCookie.parser ~ (string("; ") *> RequestCookie.parser).rep0).map {
-      case (head, tail) =>
-        Cookie(NonEmptyList(head, tail))
+    val cookieString = (RequestCookie.parser ~ (
+      (char(';') *> char(' ').rep0).soft *> RequestCookie.parser
+    ).rep0).map { case (head, tail) =>
+      Cookie(NonEmptyList(head, tail))
     }
 
     // We also see trailing semi-colons in the wild, and grudgingly tolerate them here
```

Comment:
Maybe we can add a comment similar to this one esp. since the implementation no longer matches the comment above.

Reply:
👍 added a comment